packages/js/product-editor/src/hooks/use-error-handler.ts

```diff
@@ -74,22 +73,29 @@ function getErrorPropsWithActions(
 }
 
 export const useErrorHandler = (): UseErrorHandlerTypes => {
-	const { focusByValidatorId } = useValidations();
-	const { getParentTabId, getParentTabIdByBlockName } = useBlocksHelper();
+	const { focusByValidatorId, getFieldByValidatorId } = useValidations();
+	const { getClientIdByField, getParentTabId, getParentTabIdByBlockName } =
+		useBlocksHelper();
+
+	async function getClientIdByValidatorId( validatorId: string ) {
+		if ( ! validatorId ) {
+			return null;
+		}
+		const field = await getFieldByValidatorId( validatorId );
+		return getClientIdByField( field );
```

Comment:
Good call! I added a check before calling `getClientIdByField` in the commit [e3fc6d2](https://github.com/woocommerce/woocommerce/pull/50035/commits/e3fc6d2592f0e67a88225fe7476ea343acbdead9)
After adding that check, I think we won't need to add the extra tests for `getFieldByValidatorId` returning `undefined`.

Reply:
> I think the types may not totally match here either cause it does expect the validatorId to be registered. While there is a very high chance this is always the case, there may be a use case where this doesn't happen?

To use the [useValidation hook](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/use-validation.ts#L14) we need to send the `validatorId`. To validate a field we will need to previesouly declare the validator. So having an undeclared `validatorId` doesn't seem possible. However, we set a default value for the `validatorId` [here](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/hooks/use-error-handler.ts#L90), and if we receive an invalid `validatorId` (like a random string), then [getFieldByValidatorId](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L84) will return `undefined`.
That shouldn't be a problem because we do other checks later.